### PR TITLE
Add focus ring/border in Windows High Contrast mode

### LINKED ISSUE #1251: [Global] Missing focus ring/border in Windows High Contrast
## Issue summary

Several UI elements are missing focus rings and/or borders when navigating in Windows High Contrast mode.

- Country/lang missing focus ring
- Submit controls missing border and focus ring
- No focus ring on inputs
- "Show now", "View all" controls
- Featured product select and cta controls do not feature focus ring
- Filter (remove) controls do not feature focus ring
- Product image controls do not feature focus ring
- Product buy/add to cart controls do not feature focus ring or border
- Product model "View my cart"/checkout controls do not feature focus ring or border
- Cart "remove" and checkout controls do not feature focus ring
- Cart "your cart is empty" control does not feature focus ring

#### CSS

```css
.field__input:focus,
.select__select:focus,
.customer .field input:focus,
.customer select:focus,
.localization-form__select:focus {
  /* … */
  outline: transparent;
}

.button:focus-visible,
.button:focus,
.button.focused,
.shopify-payment-button__button--unbranded:focus-visible,
.shopify-payment-button [role='button']:focus-visible,
.shopify-payment-button__button--unbranded:focus,
.shopify-payment-button [role='button']:focus {
  outline: 0;
  /* … */
}

.button, .shopify-challenge__button, .customer button {
  /* … */
  border: 0;
  /* … */
}

.customer .field input:focus, .customer select:focus, .field__input:focus, .localization-form__select:focus, .select__select:focus {
  /* … */
  outline: transparent;
  /* … */
}

a.active-facets__button:focus-visible {
  outline: none;
  /* … */
}

.product__media-toggle:focus-visible {
  outline: 0;
  /* … */
}

.thumbnail[aria-current]:focus, .thumbnail.focused {
  outline: 0;
  /* … */
}
```

## Steps to reproduce

1. Go to a demo store with Edge browser in Windows
2. With an active High Contrast theme, <kbd>Tab</kbd> through and review focus rings and borders of interactive controls

### Expected

- Focus rings and borders to be visible in High Contrast theme.

### Actual

- Focus rings and borders are not visible.

## Recommendation

For applicable CSS definitions:

1. Adjust `outline: none;` or `outline: 0;` to `outline: transparent solid 1px;`
2. Adjust `outline: transparent;` to `outline: transparent solid 1px;`
3. Adjust `border: 0;` to `border: transparent solid 1px;`

These changes will provide a focus ring/border to the appripriate element in Windows High Contrast themes only.

#### CSS

```css
.field__input:focus,
.select__select:focus,
.customer .field input:focus,
.customer select:focus,
.localization-form__select:focus {
  /* … */
  outline: transparent solid 1px;
}

.button:focus-visible,
.button:focus,
.button.focused,
.shopify-payment-button__button--unbranded:focus-visible,
.shopify-payment-button [role='button']:focus-visible,
.shopify-payment-button__button--unbranded:focus,
.shopify-payment-button [role='button']:focus {
  outline: solid transparent 1px;
  /* … */
}

.button, .shopify-challenge__button, .customer button {
  /* … */
  border: transparent solid 1px;
  /* … */
}

.customer .field input:focus, .customer select:focus, .field__input:focus, .localization-form__select:focus, .select__select:focus {
  /* … */
  outline: transparent solid 1px;
  /* … */
}
a.active-facets__button:focus-visible {
  outline: transparent solid 1px;
  /* … */
}

.product__media-toggle:focus-visible {
  outline: transparent solid 1px;
  /* … */
}

.thumbnail[aria-current]:focus, .thumbnail.focused {
  outline: transparent solid 1px;
  /* … */
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 2.5.0
- **Browser:** Any
- **Component:** Global
- **WCAG Principle:** Perceivable
- **WCAG SC:** [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
- **Severity:** High
- **Effort:** Low

### PR DESCRIPTION
**Pull request by pizzaz93**
Fixes #1251

### CODE REVIEW

**Comment by pizzaz93** [comment]
This suggestion is repeated twice in the suggestions and `.localization-form__select:focus` is now in a separate styling block.

```
.field__input:focus,
.select__select:focus,
.customer .field input:focus,
.customer select:focus,
.localization-form__select:focus {
  /* … */
  outline: transparent solid 1px;
}

```

**Comment by ludoboludo** [comment]
I think that'd make sense 🤔 and feels like a good way to go about it 👍

**Comment by pizzaz93** [comment]
I'm not sure if I implemented the nested media query correctly.

File: assets/section-main-product.css [hunk 1]
```diff
@@ -1100,6 +1101,7 @@ a.product__text {
 .thumbnail.focused {
   outline: 0;
   box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),0 0 0rem 0.5rem rgba(var(--color-foreground), 0.5);
+  @media (forced-colors: active) { outline: transparent solid 1px; }
```
[the remark this answers is not in the document]
Comment (pizzaz93) [reply assets/section-main-product.css]: Implemented

**Comment by ludoboludo** [comment]
Oh also I think it would need to be added to the gift card CSS as well. `template-giftcard.css`.

Maybe just applied there for `*:focus-visible`.

File: assets/base.css [hunk 2]
```diff
@@ -1314,6 +1314,36 @@ button.shopify-payment-button__button--unbranded {
...
+  .customer .field input:focus,
+  .customer select:focus,
+  .localization-form__select:focus.localization-form__select:after {
+    outline: transparent solid 1px; 
+  }
+
+  .localization-form__select:focus {
+    outline: transparent solid 1px; 
+  }
+}
```
[the remark this answers is not in the document]
Comment (pizzaz93) [reply assets/base.css]: Moved to the bottom of the file.

**Comment by pizzaz93** [comment]
> Oh also I think it would need to be added to the gift card CSS as well. `template-giftcard.css`.
> 
> Maybe just applied there for `*:focus-visible`.

What styling would it need to be?

Would it be adding it to this selector https://github.com/pizzaz93/dawn/blob/4319360ad3705900fd5476062167e47ed598728a/assets/template-giftcard.css#L345?

**Review by ludoboludo — APPROVED** [review]
Thanks for another useful PR @pizzaz93! 🎉

**Review by andrewetchen — APPROVED** [review]
Thanks again, @pizzaz93 - everyone will really appreciate these fixes.

I tested things on Microsoft Edge running on a Windows 11 Enterprise VM and made sure to test all [listed cases](https://github.com/Shopify/dawn/issues/1251#:~:text=Country/lang%20missing%20focus%20ring,does%20not%20feature%20focus%20ring) and everything is looking great! 🎉 

Here are some screenshots using the Aquatic [high contrast theme](https://docs.microsoft.com/en-us/windows/apps/design/accessibility/high-contrast-themes) now showing the focus rings where before they didn't show up at all:

- [Email input](https://user-images.githubusercontent.com/28404165/174896822-6e56477f-15d7-4d6f-8d0a-67bcd17889eb.png)
- [Add to cart button](https://user-images.githubusercontent.com/28404165/174896831-236ec5f3-70c1-474c-a8fd-ec8c0af186e0.png)

**Review by ludoboludo — APPROVED** [review]

**Pull request merged**